### ISSUE TITLE
F821 when using `global` 

### ISSUE THREAD

**Issue by patrick91** [charliermarsh/ruff#960]
I'm getting this error when running ruff on a code like this:

```
import textwrap
from typing import List

import strawberry


def test_entities_type_when_no_type_has_keys():
    global Review

    @strawberry.federation.type
    class User:
        username: str

    @strawberry.federation.type(extend=True)
    class Product:
        upc: str = strawberry.federation.field(external=True)
        reviews: List["Review"]
```

(we use global as a work-around for some typing issues) :)

This is the codebase I'm using Ruff against: https://github.com/strawberry-graphql/strawberry/pull/2367

**Comment by charliermarsh** [comment]
I'll look into this and see what I can do :)

**Comment by charliermarsh** [comment]
I guess the issue is that `del Review` removes `Review` from the scope, and so when we go to evaluate the forward annotation, it's gone.

Could I help with removing the `global` / `del`? How are you testing that code?

**Comment by patrick91** [comment]
The code needs global to allow get_type_hints to work at runtime, since it tries to find the forward ref from the global scope[1].

Maybe I can remove the `del`, but in theory the code is fine, no? 

For now, I disabled that error for those files though, I'm fine with ignoring it for those tests 😊

[1] Reference: https://mail.python.org/archives/list/typing-sig@python.org/thread/SNKJB2U5S74TWGDWVD6FMXOP63WVIGDR/#A5JFR7RDE7NNKESM2DWYM3ZT5ZYGOLCF

**Comment by charliermarsh** [comment]
This actually got fixed in #1154!
